dev-docs/source/DesignDocumentGuides.rst

```diff
@@ -10,39 +10,96 @@ Design Document Guidelines
 A good design document minimises unexpected complications by addressing
 them before the code is written. A document will provide you, your
 manager and your team with a common vocabulary for talking about the
-project. Writing an effective design document can be tricky in itself
+project. Writing an effective design document can be tricky in itself, as
 they often add unnecessary complexity if not done with care. These
-guidelines are to help you generate effective and simple design
+guidelines are to help when and how to generate effective and simple design
 documents.
 
-Before starting, all Mantid design documents must be written in
-`markdown <http://en.wikipedia.org/wiki/Markdown>`__ format with the md
-extension. After approval from the the Mantid Project :ref:`TSC`, they should be stored
-`here <https://github.com/mantidproject/documents/tree/master/Design>`__
-or in one of the subdirectories of that directory. This page covers what
-happens between those points.
+When to write a design document
+###############################
 
-Guidelines on Sections
-######################
+This section provides some guidelines on when to produce a design document.
+Note that, it is not about when to do a design and when not.
+One should always do design work before coding, no matter the scope and the nature of the feature.
+These guidelines are to answer in which cases the design must be a *public*, *written* and *persistent* document, rather than a sketch on a white-board in a stand-up meeting that is wiped-out after it has been turned into code.
+Just as coding, design is an iterative and collaborative process.
+Having the document written allows for discussion and iterations with peers before the first line of code is written.
+Besides, the written document facilitates the pull request review process.
+Having the design already approved, the reviewer will not need to do a design review with the code review, but will just need to make sure that the implementation matches the previously agreed design.
+Finally, once written, the document can also serve as (or easily be turned into) a developer documentation for future reference.
+
+Not every feature requires a design document, as the benefit it adds must be gaged with the time and effort that goes into it.
+Bug fixes, small extensions to the current functionalities, as well as concrete, single-class plugins, such as algorithms, do not need a document.
+In the contrary, if the feature is large-scope, brand-new, many-class, including abstractions, and/or touches or refactors the internals of the core framework, it will most likely need a document.
+Below are a couple of metrics that could help a developer decide whether or not the feature falls into that category.
+
+- Number of classes and components.
+
+  If the number of classes is more than just a few, the document is relevant in order to describe the relations between them.
```

Comment:
I feel that few is a subjective term here and can be interpreted very differently by different people. Can we express this more clearly?